### PR TITLE
On *NIX systems, gather cross compiling info from sysconfig

### LINKED ISSUE #138: Get --target, etc., from sysconfig when cross compiling
I am the maintainer of [crossenv](https://github.com/benfogle/crossenv), and since `cryptography` started using `setuptools-rust`, I've started receiving issues since cross compiling that package no longer works out of the box. This can be fixed by gathering more information from `sysconfig` and passing that along to rustc.

(PR incoming)

### PR DESCRIPTION
**Pull request by benfogle**
Closes #138

### CODE REVIEW

File: setuptools_rust/build.py [hunk 1]
```diff
@@ -112,12 +170,15 @@ def build_extension(self, ext: RustExtension, target_triple=None):
                 "PYO3_PYTHON": os.environ.get("PYO3_PYTHON", sys.executable),
             }
         )
+
+        if target_info.cross_lib:
+            env.setdefault("PYO3_CROSS_LIB_DIR", target_info.cross_lib)
```
Comment (messense) [inline setuptools_rust/build.py:175]: 👍

**Comment by davidhewitt** [comment]
Thanks! I'm going to review this together with #134, as they look like two complimentary pieces of the puzzle to me.

I confess I haven't tried cross-compiling Python projects. Have you got a recommended target to cross compile to which I can use to play around with this?

**Comment by benfogle** [comment]
Well, buildroot is a pretty straightforward way of generating Python for cross compiling. Otherwise, I have some prebuilt ones that I use for testing [here](https://github.com/benfogle/crossenv/blob/master/tests/prebuilt/prebuilt_musl_arm_aarch64.tar.xz) and [here](https://github.com/benfogle/crossenv/blob/master/tests/prebuilt/prebuilt_python3.9.0.tar.xz) for aarch64-linux-musl and arm-linux-musl.

**Review by davidhewitt — APPROVED** [review]
Thanks very much for this. Sorry for the delayed review; I've had to do a fair bit of reading on crossenv and cross-compiling Python to give this review proper attention.

It looks to me like this change is beneficial in general (not just for crossenv) because reading sysconfig is generally a good idea.

I did a force-push to rebase and add a changelog entry.

**Pull request merged**